---
title: "ATAC_3_diffexpression"
output: github_document
---

```{r setup, include=FALSE}

knitr::opts_chunk$set(echo = TRUE)

```

# ATAC analysis
Differential expression between gates, conditions and timepoints

```{r message=FALSE}

rm(list=ls())

library(DESeq2)
library(RColorBrewer)
library(tidyverse)
library(ComplexHeatmap)
library(tximport)
library(ggrepel)

```


### Set dirs
```{r}

workingdir="~/Dropbox (The Francis Crick)/DP_cisReg/"
subworkinput="inputs_CaTSATAC/"
outdir="outputs_CaTSATAC_3_diffexpression/"

ifelse(!dir.exists(file.path(workingdir,outdir)), dir.create(file.path(workingdir,outdir)), "Directory exists")

suboutdir1="output_between_gates/"
ifelse(!dir.exists(file.path(workingdir,outdir,suboutdir1)), dir.create(file.path(workingdir,outdir,suboutdir1)), "Directory exists")
suboutdir2="output_between_timepoints/"
ifelse(!dir.exists(file.path(workingdir,outdir,suboutdir2)), dir.create(file.path(workingdir,outdir,suboutdir2)), "Directory exists")
suboutdir3="output_between_conditions/"
ifelse(!dir.exists(file.path(workingdir,outdir,suboutdir3)), dir.create(file.path(workingdir,outdir,suboutdir3)), "Directory exists")
suboutdir4="output_between_gates_ignoringothers/"
ifelse(!dir.exists(file.path(workingdir,outdir,suboutdir4)), dir.create(file.path(workingdir,outdir,suboutdir4)), "Directory exists")


```



## Load data

Load the diff expression from previous script.


## Colors and shapes

```{r }
sorted_gate <- c("pMN","DP","p3","neurons")
sorted_conditions <- c("500","UPSAG","dRA2UPSAG","dRA")


shapes4_manual = c(18,15,16,17) # these are block
shapes5_manual = c(25,21,22,23,24) # these are filled
shapes4_fill_manual = c(23,21,22,24)
 

color_gates <- c("#e60000","#cd00cd","#0073e6","#696969")

# for Days
colors_greys <- c("#f6f6f6","#808080","#333333")

# conditions

colors_conditions <- c("#e67300","#4d9a00","#cdcd00","#0073e6")
```


##Load vsd to plot heatmaps later


```{r }

count_vsd <- read.csv(file=paste0(workingdir,"outputs_CaTSATAC_3_diffexpression/","consensus_peaks.mLb.clN.featureCounts.csv"),header=TRUE, stringsAsFactors = FALSE)

dds_counts <- read.table(file=paste0(workingdir,"outputs_CaTSATAC_3_diffexpression/","consensus_peaks.mLb.clN.featureCounts.txt"),header=TRUE, stringsAsFactors = FALSE)

```


Differential analysis between domains for a given timepoint and condition

• For a given timepoint and a given condition
• Differential gene expression between domains

```{r }

PairWiseDEseq_domain <- lapply(list.files(path=paste0(workingdir,subworkinput,suboutdir1),pattern="Results_DESeq*", full.names=FALSE),function(x) {
  data <- read.table(paste0(workingdir,subworkinput,suboutdir1,x),header=T,stringsAsFactors=F) %>% as.data.frame() %>% rownames_to_column("Geneid")
  data$Comparison <- gsub("Results_DESeq_","", x)
  data$Comparison <- gsub(".txt","",data$Comparison)
  data
})

results_deseq_domain <- do.call(rbind,PairWiseDEseq_domain)

```



Differential analysis between timepoints for each domain and condition
• For a given domain and a given condition
• Differential gene expression between timepoints


```{r }


PairWiseDEseq_days <- lapply(list.files(path=paste0(workingdir,subworkinput,suboutdir2),pattern="Results_DESeq*", full.names=FALSE),function(x) {
  data <- read.table(paste0(workingdir,subworkinput,suboutdir2,x),header=T,stringsAsFactors=F) %>% as.data.frame() %>% rownames_to_column("Geneid")
  data$Comparison <- gsub("Results_DESeq_","", x)
  data$Comparison <- gsub(".txt","",data$Comparison)
  data
})

results_deseq_days <- do.call(rbind,PairWiseDEseq_days)
```


Differential analysis between conditions for each domain and timepoint
• For a given timepoint and a given domain
• Differential gene expression between conditions


```{r }

PairWiseDEseq_conditions <- lapply(list.files(path=paste0(workingdir,subworkinput,suboutdir3),pattern="Results_DESeq*", full.names=TRUE),function(x) {
  data - read.table(x,header=T,stringsAsFactors=F) %% as.data.frame() %>% rownames_to_column("Geneid")
  data$Comparison <- gsub(paste0("/Users/delasj/Dropbox \\(The Francis Crick\\)/DP_cisReg/",subworkinput,suboutdir3,"/Results_DESeq_"),"", x)
  data$Comparison <- gsub(".txt","",data$Comparison)
  data
})

results_deseq_conditions <- do.call(rbind,PairWiseDEseq_days)
```


How many diff acc elements between domains?

```{r }

adjusted_pval = 0.05

log2FC = 1

minBaseMean = 80

```

```{r }

top_domain_comparisons <- results_deseq_domain %>%
  as.data.frame() %>%
  filter(padj < adjusted_pval & abs(log2FoldChange) > log2FC & baseMean > minBaseMean)

 
ggplot(top_domain_comparisons, aes(x=Comparison)) +
  geom_bar() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))

```


Are they the same ones?

```{r }

comparison_vector <- top_domain_comparisons$Comparison %% unique()

list_test <- lapply(comparison_vector, function(x) {
  top_domain_comparisons[top_domain_comparisons$Comparison==x,"Geneid"]
})
names(list_test) <- comparison_vector

#upset(fromList(list_test), sets=comparison_vector, order.by = "degree")
upset(fromList(list_test), sets=comparison_vector, order.by = "freq")
```

